### ISSUE TITLE
ProtocolLib for CatServer, Magma and Mohist (Forge Support)

### ISSUE THREAD

**Issue by red0fireus** [dmulloy2/ProtocolLib#825]
Hello, I know this feature hasn't been mentioned since [ProtocolLib 3.6.4](https://www.spigotmc.org/resources/protocollib.1997/update?update=34124) but I was wondering would it be possible to receive an update that makes ProtocolLib compatible with a forge server again? 

Link:
[CatServer](https://github.com/Luohuayu/CatServer)
[Magma](https://github.com/magmafoundation/Magma)
[Mohist](https://github.com/Mohist-Community/Mohist)

The version of Minecraft that these servers are running on is 1.12.2 and they support more **spigot and bukkit plugins** then the old MCPC/Cauldron. I was wondering if some king of support could be added back?

Thank you. (The reason for my Request is that **(I AM)** receiving errors while using protocolib with these versions of Minecraft and that is why I'm posting about this)

**Comment by red0fireus** [comment]
One more thing, I see that there was a forum thread about Magma/CatServer awhile back (August 2019).
But it looks like nothing was done to update the plugin. I was wondering if there has been any progress?
https://github.com/dmulloy2/ProtocolLib/issues/676

**Comment by MiniDigger** [comment]
protocollib doesnt support forge, this is out of scope, and such these hacks dont work

PRs are welcome tho I guess

**Comment by SauROnmiKE** [comment]
I would also like to see Forge support again, Magma has so many possibilities, but I know it is difficult to tackle as well.

**Comment by dmulloy2** [comment]
It doesn't seem like the MCPC/Cauldron successors have much traction. Same goes for Sponge, for that matter. Doesn't seem super worth it, at least at this point, to support it.

**Comment by dmulloy2** [comment]
@red0fireus give the latest build a shot and let me know if it works better

**Comment by red0fireus** [comment]
I tried the most recent Dev Version. It allows me to connect to the server now but plugins don't seem to register well with ProtocolLib. I noticed this with Lockette and Orebfuscator. Both plugins see ProtocolLib and hook in to the plugin but don't utilize the plugin. 

Here is an error I have with a plugin.

**Comment by red0fireus** [comment]
Here's a better view of the error: 
https://pastebin.com/T7aGjZLX

**Comment by MiniDigger** [comment]
The plugin is incompatible with your minecraft version

**Comment by dmulloy2** [comment]
It does look like ProtocolLib itself is working, so I'm going to count this as resolved on our end.

**Comment by NewbieOrange** [comment]
Hi, I found that although ProtocolLib does load and seemingly work, there are issues identifying PacketType when receiving packets (sending not tested). This caused issues with plugins. I have found that IllegalStacks and FastLogin have encountered the issue.

Here is the stack trace from FastLogin: https://pastebin.com/sYJGCyup

From the debug message at first line, we can see that the packet "class" is: `[class=PacketLoginInStart, id=0]`

However, in the code, the condition `if (packetEvent.getPacketType() packetType == START)` evaluated to false and went to the else branch, trying to parse the packet as an `encryption_begin` packet, causing errors.

I understand that forge servers are not supposed to be supported, but it would be great if ProtocolLib works. 😃
